src/platform/common/utils/localize.ts

```diff
@@ -244,27 +271,36 @@ export namespace DataScience {
         'Failed to connect to password protected server. Check that password is correct.'
     );
     export const rawKernelProcessNotStarted = localize(
-        'DataScience.rawKernelProcessNotStarted',
+        { key: 'DataScience.rawKernelProcessNotStarted', comment: ['{Locked="kernel"}'] },
         'Raw kernel process was not able to start.'
     );
     export const rawKernelProcessExitBeforeConnect = localize(
-        'DataScience.rawKernelProcessExitBeforeConnect',
+        { key: 'DataScience.rawKernelProcessExitBeforeConnect', comment: ['{Locked="kernel"}'] },
         'Raw kernel process exited before connecting.'
     );
     export const unknownMimeTypeFormat = localize(
-        'DataScience.unknownMimeTypeFormat',
+        { key: 'DataScience.unknownMimeTypeFormat', comment: ['{Locked="Mime type"}'] },
         'Mime type {0} is not currently supported'
     );
-    export const exportDialogTitle = localize('DataScience.exportDialogTitle', 'Export to Jupyter Notebook');
-    export const exportDialogFilter = localize('DataScience.exportDialogFilter', 'Jupyter Notebooks');
+    export const exportDialogTitle = localize(
+        { key: 'DataScience.exportDialogTitle', comment: ['{Locked="Notebook"}'] },
+        'Export to Jupyter Notebook'
+    );
+    export const exportDialogFilter = localize(
+        { key: 'DataScience.exportDialogFilter', comment: ['{Locked="Notebooks"}'] },
+        'Jupyter Notebooks'
+    );
     export const exportDialogComplete = localize('DataScience.exportDialogComplete', 'Notebook written to {0}');
     export const exportDialogFailed = localize('DataScience.exportDialogFailed', 'Failed to export notebook. {0}');
     export const exportOpenQuestion1 = localize('DataScience.exportOpenQuestion1', 'Open in editor');
     export const runCellLensCommandTitle = localize('jupyter.command.jupyter.runcell.title', 'Run Cell');
-    export const importDialogTitle = localize('DataScience.importDialogTitle', 'Import Jupyter Notebook');
-    export const importDialogFilter = localize('DataScience.importDialogFilter', 'Jupyter Notebooks');
+    export const importDialogTitle = localize(
+        { key: 'DataScience.importDialogTitle', comment: ['{Locked="Notebook"}'] },
```

Comment:
I would have thought the name of the product `Jupyter`, `Python` & others, too need locked and should not be localized.

Reply:
In both cases, they're not real words, so I would not assume someone would translate them. Adding a lock for them is relatively easy, but it means many more changes that may be unnecessary. We need to check the translations before we release it, so if we see something weird going on when the translations are ready, we could add the locks. How does that sound?